packages/@aws-cdk/aws-codepipeline/lib/action.ts

```diff
@@ -27,12 +27,29 @@ export interface ActionArtifactBounds {
   readonly maxOutputs: number;
 }
 
-export function defaultBounds(): ActionArtifactBounds {
+/**
+ * The ArtifactBounds that make sense for source Actions -
+ * they don't have any inputs, and have exactly one output.
+ */
+export function sourceArtifactBounds(): ActionArtifactBounds {
   return {
     minInputs: 0,
-    maxInputs: 5,
+    maxInputs: 0,
+    minOutputs: 1,
+    maxOutputs: 1,
+  };
+}
+
+/**
+ * The ArtifactBounds that make sense for deploy Actions -
+ * they have exactly one input, and don't produce any outputs.
+ */
+export function deployArtifactBounds(): ActionArtifactBounds {
```

Comment:
This isn't true (I can create deploy actions with output artifacts) and artifact bounds are set per-action type. Let's minimally move this to aws-codepipeline-actions as a helper function or ideally let each action type declare its own artifact bounds.

Reply:
Moved these 2 to `aws-codepipeline-actions` and made them private.